pkg/operator/ceph/cluster/controller.go

```diff
@@ -498,6 +533,31 @@ func (c *ClusterController) initializeCluster(cluster *cluster, clusterObj *ceph
 // ************************************************************************************************
 // Update event functions
 // ************************************************************************************************
+func (c *ClusterController) rookCephOperatorConfigUpdate(oldObj, newObj interface{}) {
+	if reflect.DeepEqual(oldObj, newObj) {
+		return
+	}
+	cm, ok := newObj.(*v1.ConfigMap)
+	if !ok {
+		logger.Warningf("Expected ConfigMap but handler received %#v", newObj)
+		return
+	}
+
+	logger.Infof("%s Config Map updated. Trying to update deployment %s", cm.Name, "rook-ceph-operator")
+	deployment, err := c.context.Clientset.AppsV1().Deployments(os.Getenv(k8sutil.PodNamespaceEnvVar)).Get("rook-ceph-operator", metav1.GetOptions{})
+	if err != nil {
+		logger.Warningf("Error fetching deployment %s. Config changes can not be applied. %v", "rook-ceph-operator", err)
+		return
+	}
+	deployment, err = c.context.Clientset.AppsV1().Deployments(os.Getenv(k8sutil.PodNamespaceEnvVar)).Update(deployment)
```

Comment:
Is your intention of updating the deployment that it will restart the operator? Nothing is changing in the deployment spec so this won't have the desired effect. While restarting the operator is expected to be ok since the operator is stateless, it still is not something I would want to build into the design.

For the approach, I was thinking that when the operator starts, each setting that supports being updated at runtime will register a callback method for applying the settings. Then when the configmap is updated, those callbacks would all be called with the new settings. For example, the CSI driver would then be updated by one of those callbacks if the CSI settings change.

Reply:
> Is your intention of updating the deployment that it will restart the operator? Nothing is changing in the deployment spec so this won't have the desired effect. While restarting the operator is expected to be ok since the operator is stateless, it still is not something I would want to build into the design.
>
Yes. It would need some work to make it restart which is not worth it.

Reply:
> For the approach, I was thinking that when the operator starts, each setting that supports being updated at runtime will register a callback method for applying the settings. Then when the configmap is updated, those callbacks would all be called with the new settings. For example, the CSI driver would then be updated by one of those callbacks if the CSI settings change.
>
Sounds good.

When the ConfigMap is changed, all the callbacks would be called. Is this true even if only some config are changed?